lib/helpers.php

```diff
@@ -149,7 +153,7 @@ function micropub_post($endpoint, $params, $access_token, $file_path = NULL, $js
     $mimetype = finfo_file($finfo, $file_path);
     $multipart = new p3k\Multipart();
     $multipart->addArray($params);
-    $multipart->addFile($file_prop, $file_path, $mimetype);
+    $multipart->addFile($file_prop, $filename, $mimetype, $file_content);
```

Comment:
That's really strange... I swear I checked that when writing this comment and didn't see the 4th parameter.